test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_CodeFlow.cs

```diff
@@ -0,0 +1,172 @@
+    {
+        _parameters = await TestParameters.GetAsync();
+        ConfigureDarcArgs();
+    }
+
+    [Test]
+    public async Task Darc_ForwardFlowTest()
+    {
+        var channelName = GetTestChannelName();
+        var branchName = GetTestBranchName();
+        var productRepo = GetGitHubRepoUrl(TestRepository.TestRepo1Name);
+        var targetBranchName = "main";
+
+        await using AsyncDisposableValue<string> testChannel = await CreateTestChannelAsync(channelName);
+
+        await using AsyncDisposableValue<string> subscriptionId = await CreateSourceEnabledSubscriptionAsync(
+            channelName,
+            TestRepository.TestRepo1Name,
+            TestRepository.VmrTestRepoName,
+            targetBranchName,
+            UpdateFrequency.None.ToString(),
+            _parameters.GitHubTestOrg,
+            [],
+            targetDirectory: TestRepository.TestRepo1Name);
+
+        TemporaryDirectory reposFolder = await CloneRepositoryAsync(TestRepository.TestRepo1Name);
+        var newFilePath = Path.Combine(reposFolder.Directory, TestFileName);
+
+        using (ChangeDirectory(reposFolder.Directory))
+        {
+            await using (await CheckoutBranchAsync(branchName))
+            {
+                // Make a change in a product repo
+                TestContext.WriteLine("Making code changes to the repo");
+                using FileStream newFileStream = File.Create(newFilePath);
+                {
+                    using StreamWriter newFileWriter = new(newFileStream);
+                    newFileWriter.Write(TestFilesContent[TestFileName]);
+                }
+
+                await GitAddAllAsync();
+                await GitCommitAsync("Add new file");
+
+                // Push it to github
+                await using (await PushGitBranchAsync("origin", branchName))
+                {
+                    var repoSha = (await GitGetCurrentSha()).TrimEnd();
+
+                    // Create a new build from the commit and add it to a channel
+                    Build build = await CreateBuildAsync(
+                        GetGitHubRepoUrl(TestRepository.TestRepo1Name),
+                        branchName,
+                        repoSha,
+                        "1",
+                        []);
+
+                    TestContext.WriteLine("Adding build to channel");
+                    await AddBuildToChannelAsync(build.Id, channelName);
+
+                    TestContext.WriteLine("Triggering the subscription");
+                    // Now trigger the subscription
+                    await TriggerSubscriptionAsync(subscriptionId.Value);
+
+                    TestContext.WriteLine("Verifying subscription PR");
+                    await CheckForwardFlowGitHubPullRequest(TestRepository.TestRepo1Name, TestRepository.VmrTestRepoName, targetBranchName, [TestFileName], TestFilePatches);
+                }
+            }
+        }
+    }
+
+    [Test]
+    public async Task Darc_BackwardFlowTest()
```

Comment:
Why `Darc_`?

Reply:
All Scenario tests have that prefix, so I just continued it. It's kind of nice when you're looking for a failed test in the logs, I usually just search for `Failed Darc_`

Reply:
Ah, my bad. I guess these never fail, so I never really looked at them. In that case maybe we should name these something like `VMR_ForwardFlow`